### DISCUSSION TITLE
Using theme with WezTerm

### DISCUSSION THREAD

**Discussion by Drllap** [Bash-it/bash-it#2237] [Q&A]
The theme doesn't kick in the first prompt :
![image](https://github.com/Bash-it/bash-it/assets/3659986/25a72578-ab10-4076-82fa-41fb28844976)
I'm using WezTerm. It starts bash in login mode, maybe it has something to do with that?
It works fine both in Alacritty and XTerm.
Should I create an issue?

**Comment by boldandbusted** [comment]
I'm not seeing this behavior with WezTerm, which is my "daily driver" terminal. Not sure the best approach to debugging this, however.

**Reply by Drllap** [reply]
I did a bit of experimenting and found out I had a similar issue with starship: https://github.com/starship/starship/issues/5686

The issue is also present when using Aalacritty/XTerm but the difference is that WezTerm starts bash in login mode by default.

WezTerm shipps with the [shell-integration file](https://github.com/wez/wezterm/blob/cce0706b1f2a9e2d1f02c57f2d1cd367c91df1ae/assets/shell-integration/wezterm.sh#L4) that is sourced when opening bash in login mode. Removing it fixes the issue

**Comment by boldandbusted** [comment]
So, I think this issue was resolved, but not sure how to mark it as "answered" here.

**Reply by akinomyoga** [reply]
The OP or a maintainer can mark the answer. By the way, how was this solved?

**Reply by Drllap** [reply]
This is not really solved but this isn't related to bash-it. This has something to do with the shell integration scripts shipped with WezTerm/bash-excec.
There are a couple of workarounds: 

- start bash in non -login mode, then those scripts will not be sourced
- edit the script so that the bash-excec isn't executed

**Reply by akinomyoga** [reply]
Thanks. So, can the problem be reproducible just with WezTerm's shell integration and bash-preexec and without Bash-it?

Hmm, is this related (where you seem to have left comments)?

https://github.com/wez/wezterm/issues/4406

In the above issue, the maintainer of WezTerm said they have worked around it, and the issue has already been closed. Does that mean the workaround was not complete?

**Reply by Drllap** [reply]
> So, can the problem be reproducible just with WezTerm's shell integration and bash-preexec and without Bash-it?

Yes, this also happens with Starship: https://github.com/starship/starship/issues/5686

If WezTerm is installed the shell integration scripts will be sourced no matter what terminal emulator you use as long as it opens bash in login mode (at least on my system)

> In the above issue, the maintainer of WezTerm said they have worked around it, and the issue has already been closed. Does that mean the workaround was not complete?

I didn't think this was fixed but it may be. If I remember correctly, I modified the shell integration scripts locally and then stopped thinking about this. I will look into this when I have time and report back

**Reply by Drllap** [reply]
Ok, I finally got around to check this again, and this is still an issue if I use the original `wezterm.sh` shell integration script shipped with WezTerm. I got it to work by commenting out these two lines [here](https://github.com/wez/wezterm/blob/30345b36d8a00fed347e4df5dadd83915a7693fb/assets/shell-integration/wezterm.sh#L490-L491)
